arbor/include/arbor/mechanism_abi.h

```diff
@@ -208,6 +215,8 @@ typedef struct arb_mechanism_type {
     arb_size_type             n_parameters;
     arb_ion_info*             ions;             // Ion properties
     arb_size_type             n_ions;
+    arb_random_variable_info* random_variables; // Random variable properties
```

Comment:
maybe, but I was under the impression that adding fields at the end would not destroy ABI?

Reply:
It's all about shoving binary blobs across an application boundary. Assume you would -- like here -- add a field
somewhere in the middle
Arbor passes the struct `ppack` of version `v(X+1)` to a mechanism `M` expecting one of `vX` essentially as 
a `void*`. Now `M` will cast that `void*` to `ppack(v=X)` and have all offsets wrong. You _can_ get away with it 
when you just tack a field to the end in this scenario (btw, this is how C people do inheritance ;)). But then the 
other direction breaks: `M` is generating the `mech_type` (above) which is then consumed by the library and 
has the same layout/slicing issues, since again, we pass all as `void*`.

Reply:
> Now M will cast that void* to ppack(v=X) and have all offsets wrong

I think that's precisely what does not happen, if you add at the end.

I agree about the other direction, though. Haven't thought about this. We could probably work around by placing the returned `arb_mechanism_type` within a default constructed one, which has the added fields set to zero. So the case library ABI >= mech ABI would work.

For mech ABI > library ABI, it will break, but is that even realistic?

Anyways, not opposed to bumping ABI version, just curious about this.

Reply:
should I bump minor or major?

Reply:
Minor